src/components/enterprise/dashboard/main-content/DashboardMainContent.jsx

```diff
@@ -1,10 +1,25 @@
-import React from 'react';
+import React, { Component } from 'react';
 
+import { LayoutContext } from '../../../common/layout';
 import { CourseEnrollments } from '../../../common/course-enrollments';
 import { DashboardSidebar } from '../sidebar';
 
-const DashboardMainContent = () => (
-  <CourseEnrollments sidebarComponent={<DashboardSidebar />} />
-);
+export default class DashboardMainContent extends Component {
```

Comment:
Could this be written as a function component, instead of class component? You'd need to use the `useContext` React hook to read from the `LayoutContext`. 

Since this component doesn't require any lifecycle methods, think we'd OK with just a single functional component here.

Reply:
What's the advantage of a function component?

Reply:
Typically, you want to prefer functional components unless the component requires things that only exist in class-based components such as state, or lifecycle methods. However, those kinds of things are also creeping into functional components now with hooks like `useState` and `useEffect`. Generally, best practice seems to prefer functional component unless you otherwise can't.

Functional components are typically easier to read/test since they're just functions, and usually are less code than class-based components as well.

Reply:
Hmm, I think I find class-based components more readable, but perhaps that the object-orientedness in me coming out.